src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -1,84 +1,94 @@
-    dict["z"] = typeof(b.z) == T ? b.z * 2 : OrderedDict{String,Any}("from" => b.z.left, "to" => b.z.right)
-    OrderedDict{String,Any}("box" => dict)
+function vertices(b::Box{T}) where {T}
+    return SVector{8, CartesianPoint{T}}(
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, +b.hZ) .+ b.origin,
+    )
 end
 
-get_x_limits(b::Box{T}) where {T} = (_left_linear_interval(b.x), _right_linear_interval(b.x))
-get_y_limits(b::Box{T}) where {T} = (_left_linear_interval(b.y), _right_linear_interval(b.y))
-get_z_limits(b::Box{T}) where {T} = (_left_linear_interval(b.z), _right_linear_interval(b.z))
+sample(b::Box) = vertices(b)
 
-function get_decomposed_surfaces(b::Box{T}) where {T}
-    xMin::T, xMax::T = get_x_limits(b)
-    yMin::T, yMax::T = get_y_limits(b)
-    zMin::T, zMax::T = get_z_limits(b)
-    tol = geom_atol_zero(T)
-    if isapprox(xMin, xMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleX(b, xMin)]
-    elseif isapprox(yMin, yMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleY(b, yMin)]
-    elseif isapprox(zMin, zMax, atol = tol)
-        return AbstractSurfacePrimitive[RectangleZ(b, zMin)]
-    else
-        return AbstractSurfacePrimitive[
-                                            RectangleX(b, xMin), RectangleX(b, xMax), 
-                                            RectangleY(b, yMin), RectangleY(b, yMax), 
-                                            RectangleZ(b, zMin), RectangleZ(b, zMax)
-                                        ]
-    end
+function surfaces(b::Box{T}) where {T}
+    vs = vertices(b)
+    return SVector{6, Quadrangle{T}}(
+        Quadrangle{T}((vs[1], vs[2], vs[3], vs[4])),
+        Quadrangle{T}((vs[5], vs[6], vs[2], vs[1])),
+        Quadrangle{T}((vs[8], vs[7], vs[6], vs[5])),
+        Quadrangle{T}((vs[6], vs[7], vs[3], vs[2])),
+        Quadrangle{T}((vs[7], vs[8], vs[4], vs[3])),
+        Quadrangle{T}((vs[8], vs[5], vs[1], vs[4])),
+    )
 end
 
-function sample(b::Box{T}, Nsamps::NTuple{3,Int} = (2,2,2)) where {T}
-    xMin::T, xMax::T = get_x_limits(b)
-    yMin::T, yMax::T = get_y_limits(b)
-    zMin::T, zMax::T = get_z_limits(b)
-    samples = [
-        CartesianPoint{T}(x,y,z)
-        for x in (Nsamps[1] ≤ 1 ? xMin : range(xMin, xMax, length = Nsamps[1]))
-        for y in (Nsamps[2] ≤ 1 ? yMin : range(yMin, yMax, length = Nsamps[2]))
-        for z in (Nsamps[3] ≤ 1 ? zMin : range(zMin, zMax, length = Nsamps[3]))
-    ]
+function distance(b::Box{T}, pt::CartesianPoint{T}) where {T}
```

Comment:
I think this should not be here. There since it is the same algorithm for a volume primitives. 
```julia
function distance(v::AbstractVolumePrimitive{T}, pt::CartesianPoint{T}) where {T}
    surfs = surfaces(v)  
    minimum(map(s -> distance(s, pt), surfs))
end
```

Reply:
Indeed, I should move this.

Reply:
This function actually does not exists anymore on the current version of my branch. 
I changed it to `distance(b::Box{T}, pt::CartesianPoint{T})`. So it is in the correct file. 
It is by the way only correct for points inside of the box. 
But we probably only need it for points inside the semiconductors in the drift it is enough for now.

Reply:
It would be nice if it could work for any point. The code I wrote for `distance(v::AbstractVolumePrimitive{T}, pt::CartesianPoint{T})` would evaluate distance to Quadrangles instead of distance to Planes so it would automatically work for all points. 
My point with `distance(v::AbstractVolumePrimitive{T}, pt::CartesianPoint{T})` is why would we write a distance function for every single volume primitive when we can just use the code I wrote in the comments. My suggestion is to delete `distance(b::Box{T}, pt::CartesianPoint{T})` and replace it by `distance(v::AbstractVolumePrimitive{T}, pt::CartesianPoint{T})` as it was before

Reply:
I see. I misunderstood your first comment. 
(I though somehow that your written function was somewhere in the file `Box.jl`..).
I changed it. 
Right, we will need a function which also works outside since we will have to deal with CSG unions&intersections&differences. But, we will only need this when we update the charge drift. 
And there we will probably to some clustering of the faces to certain volumes to reduce the number of distances 
we have to calculate during the drift.